constants.toml

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Shared constants for GitHub workflows and repository automation.
+
+# Rust toolchain versions
+RUST_MSRV = "1.88"                                  # used for testing (ensures the MSRV promise is kept)
+RUST_LATEST = "1.92"                                # used for static analysis & mutation testing; must match rust-toolchain.toml
+RUST_NIGHTLY = "nightly-2025-11-20"                 # used for coverage and extended analysis
+RUST_NIGHTLY_EXTERNAL_TYPES = "nightly-2025-08-06"  # used for external type exposure checks
+
+# Tool versions
+CARGO_AUDIT_VERSION = "0.22.0"
```

Comment:
Will dependabot know how to recognize these values and upgrade them automatically when needed, or does this defeat it?

Reply:
Ah, great question! I have minimal understanding of how Dependabot works and struggle to find useful evidence from its logs but from what documentation and AI insights suggest, Dependabot relies on having literal version strings and will not process anything with variables or other types of indirection.

In context of this PR, I think nothing changes - no new variables were created. All these things were indirectly specified before, all that changed was that the variables became reusable. From what my brief research tells me, these variables were not Dependabot-compatible even before this PR.